src/routing_client.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2015 MaidSafe.net limited
+// This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+// By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+// directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+// available at: http://www.maidsafe.net/licenses
+// Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied.
+// See the Licences for the specific language governing permissions and limitations relating to
+// use of the MaidSafe
+// Software.                                                                 
+
+extern crate crust;
+extern crate sodiumoxide;
+extern crate cbor;
+extern crate maidsafe_types;
+
+use sodiumoxide::crypto;
+use maidsafe_types::Random;
+use std::sync::mpsc::{Receiver, Sender};
+use std::sync::mpsc;
+use std::net::{TcpStream};
+use types;
+use Facade;
+use message_header;
+use messages;
+use maidsafe_types::traits::RoutingTrait;
+
+/// DHT node
+pub struct RoutingClient<'a> {
+    facade: &'a (Facade + 'a),
+    sign_public_key: crypto::sign::PublicKey,
+    sign_secret_key: crypto::sign::SecretKey,
+    encrypt_public_key: crypto::asymmetricbox::PublicKey,
+    encrypt_secret_key: crypto::asymmetricbox::SecretKey,
```

Comment:
Need to keep a copy of maid id here. (https://github.com/dirvine/maidsafe_types/blob/master/src/id/maid.rs) 
This should be passed in as a parameter.
No need to have other keys.